lib/rules/no-loop-func.js

```diff
@@ -48,9 +59,12 @@ function getContainingLoopNode(node) {
             case "FunctionExpression":
             case "FunctionDeclaration":
 
-                // We don't need to check nested functions.
-                return null;
+                // We need to check nested functions only in case of IIFE.
+                if (isIIFE(parent) && SKIPPED_IIFE_NODES.has(parent)) {
```

Comment:
```suggestion
                if (SKIPPED_IIFE_NODES.has(parent)) {
```

There's no need to check again if it's an IIFE when it's skipped as an IIFE.